tests/devices/test_default_qutrit_mixed.py

```diff
@@ -1270,3 +1270,320 @@ def qfunc(x, y):
+            )
+
+        res = circuit()
+        assert np.allclose(res, expected)
+
+    state_relax_and_misclass = [
+        [(0, 0, 0), (0, 0, 0)],
+        [(0.1, 0.15, 0.25), (0.1, 0.15, 0.25)],
+        [(1, 0, 1), (1, 0, 0)],
+    ]
+
+    @pytest.mark.parametrize("relaxations, misclassifications", state_relax_and_misclass)
+    def test_readout_state(self, nr_wires, relaxations, misclassifications):
+        """Tests the state output is not affected by readout error"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            wires=nr_wires,
+            readout_relaxation_probs=relaxations,
+            readout_misclassification_probs=misclassifications,
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            self.setup_state(nr_wires)
+            return qml.state()
+
+        res = circuit()
+        assert np.allclose(res, self.get_expected_dm(nr_wires))
+
+    @pytest.mark.parametrize("relaxations, misclassifications", state_relax_and_misclass)
+    def test_readout_density_matrix(self, nr_wires, relaxations, misclassifications):
+        """Tests the density matrix output is not affected by readout error"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            wires=nr_wires,
+            readout_relaxation_probs=relaxations,
+            readout_misclassification_probs=misclassifications,
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            self.setup_state(nr_wires)
+            return qml.density_matrix(wires=1)
+
+        res = circuit()
+        assert np.allclose(res, self.get_expected_dm(1))
+
+    @pytest.mark.parametrize(
+        "relaxations, misclassifications, expected",
+        [
+            ((0, 0, 0), (0, 0, 0), [(np.ones(2) * 2)] * 2),
+            (None, (0, 0, 1), [np.ones(2)] * 2),
+            ((0, 0, 1), None, [np.ones(2)] * 2),
+            (None, (0, 1, 0), [np.zeros(2)] * 2),
+            ((0, 1, 0), None, [np.zeros(2)] * 2),
+        ],
+    )
+    def test_readout_sample(self, nr_wires, relaxations, misclassifications, expected):
+        """Tests the sample output with readout error"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            shots=2,
+            wires=nr_wires,
+            readout_relaxation_probs=relaxations,
+            readout_misclassification_probs=misclassifications,
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.QutritBasisState([2] * nr_wires, wires=range(nr_wires))
+            return qml.sample(wires=[0, 1])
+
+        res = circuit()
+        assert np.allclose(res, expected)
+
+    @pytest.mark.parametrize(
+        "relaxations, misclassifications, expected",
+        [
+            ((0, 0, 0), (0, 0, 0), {"22": 100}),
+            (None, (0, 0, 1), {"11": 100}),
+            ((0, 0, 1), None, {"11": 100}),
+            (None, (0, 1, 0), {"00": 100}),
+            ((0, 1, 0), None, {"00": 100}),
+        ],
+    )
```

Comment:
**suggestion**: have at least one test that computes non-trivial statistics

Setting the probabilities to 0/1 is a good way of testing that the readout error is applied correctly :+1: It would be great to have at least one integration test that checks samples/counts give approximately correct results for arbitrary values of the probabilities, so that we can verify the distributions of readout errors are correct.

Reply:
Would it be ok to have a separate test for this so not all of them have to have a high number of shots?

Reply:
Yes of course, that would be preferable.